Library/Homebrew/dev-cmd/audit.rb

```diff
@@ -910,13 +912,17 @@ def audit_revision_and_version_scheme
           newest_committed_version ||= previous_version
           newest_committed_checksum ||= previous_checksum
           newest_committed_revision ||= previous_revision
+          newest_committed_url ||= stable.url
         end
 
         break if previous_version && current_version != previous_version
       end
 
+      pypi_prefix = "https://files.pythonhosted.org/packages/"
+
       if current_version == previous_version &&
-         current_checksum != newest_committed_checksum
+         current_checksum != newest_committed_checksum &&
+         !(newest_committed_url.start_with?(pypi_prefix) ^ current_url.start_with?(pypi_prefix))
```

Comment:
Do we need to restrict this to PyPI URLs? I would think that switching URLs any time might mean a different sha256

If so, how about:

```suggestion
         current_checksum != newest_committed_checksum &&
         newest_committed_url.start_with?(pypi_prefix) != current_url.start_with?(pypi_prefix)
```

That might be just a personal preference but I think that's easier to read than negating an xor.

Reply:
We probably want to be more conservative and just do it for PyPI urls for now until a use case arises for other domains. `!=` does read more clearly than negating an XOR, so I'll make that change.